classes/sdimg.bbclass

```diff
@@ -100,20 +103,88 @@ IMAGE_CMD_sdimg() {
+
+    PART1_START_SECTORS=$ALIGNMENT_SECTORS
+    PART1_SIZE_BYTES_UNALIGNED=$(stat -L -c %s ${WORKDIR}/fat.dat)
+    PART1_SIZE_BYTES=$(expr \( $PART1_SIZE_BYTES_UNALIGNED + $ALIGNMENT_BYTES - 1 \) / $ALIGNMENT_BYTES \* $ALIGNMENT_BYTES)
+    PART1_END_SECTORS=$(expr $PART1_START_SECTORS + $PART1_SIZE_BYTES / 512 - 1)
+
+    PART2_START_SECTORS=$(expr $PART1_END_SECTORS + 1)
+    PART2_SIZE_BYTES_UNALIGNED=$(stat -L -c %s ${WORKDIR}/part1.tmp)
+    PART2_SIZE_BYTES=$(expr \( $PART2_SIZE_BYTES_UNALIGNED + $ALIGNMENT_BYTES - 1 \) / $ALIGNMENT_BYTES \* $ALIGNMENT_BYTES)
+    PART2_END_SECTORS=$(expr $PART2_START_SECTORS + $PART2_SIZE_BYTES / 512 - 1)
+
+    PART3_START_SECTORS=$(expr $PART2_END_SECTORS + 1)
+    PART3_SIZE_BYTES_UNALIGNED=$(stat -L -c %s ${WORKDIR}/part2.tmp)
+    PART3_SIZE_BYTES=$(expr \( $PART3_SIZE_BYTES_UNALIGNED + $ALIGNMENT_BYTES - 1 \) / $ALIGNMENT_BYTES \* $ALIGNMENT_BYTES)
+    PART3_END_SECTORS=$(expr $PART3_START_SECTORS + $PART3_SIZE_BYTES / 512 - 1)
+
+    dd if=/dev/zero of=${WORKDIR}/data.ext3 count=0 bs=1M seek=${SDIMG_DATA_PART_SIZE_MB}
+    mkfs.ext3 -F ${WORKDIR}/data.ext3 -d ${WORKDIR}/data
+
+    PART4_START_SECTORS=$(expr $PART3_END_SECTORS + 1)
+    PART4_SIZE_BYTES_UNALIGNED=$(stat -L -c %s ${WORKDIR}/data.ext3)
+    PART4_SIZE_BYTES=$(expr \( $PART4_SIZE_BYTES_UNALIGNED + $ALIGNMENT_BYTES - 1 \) / $ALIGNMENT_BYTES \* $ALIGNMENT_BYTES)
+    PART4_END_SECTORS=$(expr $PART4_START_SECTORS + $PART4_SIZE_BYTES / 512 - 1)
+
+    dd if=/dev/zero of=$SDIMG count=0 seek=$(expr $PART4_END_SECTORS + 1)
+    (
+        # Create DOS partition table
+        echo o
+        # 1st partition (FAT32)
+        echo n
+        echo p
+        echo 1
+        echo $PART1_START_SECTORS
+        echo $PART1_END_SECTORS
+        # 2nd partition (1st rootfs)
+        echo n
+        echo p
+        echo 2
+        echo $PART2_START_SECTORS
+        echo $PART2_END_SECTORS
+        # 3rd partition (2nd root)
+        echo n
+        echo p
+        echo 3
+        echo $PART3_START_SECTORS
+        echo $PART3_END_SECTORS
+        # 3rd partition (2nd root)
+        echo n
+        echo p
+        echo 4
+        echo $PART4_START_SECTORS
+        echo $PART4_END_SECTORS
+        # 1st partition: bootable
+        echo a
+        echo 1
+        # 1st partition: type W95 FAT16 (LBA)
+        echo t
+        echo 1
+        echo e
+        # 2nd partition: type Linux
+        echo t
+        echo 2
+        echo 83
+        # 3rd partition: type Linux
+        echo t
+        echo 3
+        echo 83
+        # COMMIT changes to image file
+        echo p
+        echo w
+
+    ) | fdisk -c=nondos -u=sectors $SDIMG
+
+    dd if=${WORKDIR}/fat.dat of=$SDIMG seek=$PART1_START_SECTORS conv=notrunc
+    dd if=${WORKDIR}/part1.tmp of=$SDIMG seek=$PART2_START_SECTORS conv=notrunc
+    dd if=${WORKDIR}/part2.tmp of=$SDIMG seek=$PART3_START_SECTORS conv=notrunc
+    dd if=${WORKDIR}/data.ext3 of=$SDIMG seek=$PART4_START_SECTORS conv=notrunc
```

Comment:
You did a nice cleanup with the names on master branch. Maybe something similar here? `BOOT_PART_START`, `INACTIVE_START`, and so on?